modules/node-shared/src/main/scala/org/tessellation/node/shared/domain/seedlist/HistoricalSnapshotOrdinalEntries.scala

```diff
@@ -0,0 +1,116 @@
+package org.tessellation.node.shared.domain.seedlist
+
+import cats.data.NonEmptyList
+import cats.syntax.all._
+
+import scala.annotation.tailrec
+import scala.util.control.NoStackTrace
+
+import org.tessellation.schema.SnapshotOrdinal
+
+import derevo.cats.{eqv, show}
+import derevo.derive
+import eu.timepit.refined.auto._
+import eu.timepit.refined.cats._
+import eu.timepit.refined.types.numeric.NonNegLong
+import fs2.data.csv.{CellDecoder, DecoderError}
+
+@derive(eqv)
+case class HistoricalSnapshotOrdinalEntries(values: Option[NonEmptyList[HistoricalSnapshotOrdinalEntry]])
+
+case class InvalidRange(msg: String) extends NoStackTrace {
+  override def getMessage: String = msg
+}
+case class InvalidRanges(msg: String) extends NoStackTrace {
+  override def getMessage: String = msg
+}
+
+object HistoricalSnapshotOrdinalEntries {
+
+  implicit val historicalSnapshotOrdinalEntriesCellDecoder: CellDecoder[Option[HistoricalSnapshotOrdinalEntries]] =
+    CellDecoder.stringDecoder.emap { s =>
+      mkSnapshotOrdinalTimeline(s)
+        .leftMap(err => new DecoderError(err.getMessage))
+        .map(_.some)
+    }
+
+  @derive(eqv, show)
+  case class RangeBoundary(start: NonNegLong, end: NonNegLong) {
+    def overlaps(other: RangeBoundary): Boolean = other.start < end
+
+    def mergeWith(other: RangeBoundary): RangeBoundary =
+      RangeBoundary(
+        start = NonNegLong.from(Math.min(start, other.start)).toOption.get,
+        end = NonNegLong.from(Math.max(end, other.end)).toOption.get
```

Comment:
Using `Option.get` is often a sign we can do better. This should be reworked.
I again question the value of using refined types for code internal to this module. I think you could use `Long` until you're ready to produce the final class.

Reply:
On the other hand if it is just `Math.min` of 2 `NonNegLong` then I think we can use `unsafe` here as `Math.min(nonNegLongA, nonNegLongB)` always returns `NonNegLong` but we just need to cast it.

Reply:
Reworked to:

```scala
  def merge(other: Range): Range = {
    val updatedStart = Math.min(start, other.start)
    val updatedEnd = Math.max(end, other.end)

    Range(
      NonNegLong.unsafeFrom(updatedStart),
      NonNegLong.unsafeFrom(updatedEnd)
    )
  }
```